File: app/src/main/java/com/app/scavenger/ForgotPassword.java
package com.app.scavenger;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.widget.EditText;
import com.google.android.material.button.MaterialButton;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;
import com.google.firebase.auth.FirebaseAuth;

public class ForgotPassword extends AppCompatActivity {

    //private static final String TAG = "FORGOT_PASSWORD";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_forgot_password);

        EditText forgot_editText = findViewById(R.id.forgot_editText);
        MaterialButton forgot_pass_button = findViewById(R.id.forgot_pass_button);
        ConnectionDetector con = new ConnectionDetector(this);

        TopToolbar topToolbar = findViewById(R.id.forgot_toolbar);
        topToolbar.setTitle("Forgot Password");

        // Disable the button on start
        forgot_pass_button.setEnabled(false);

        // Check if text is empty or not
        forgot_editText.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                if (s.toString().trim().length() != 0) {
                    forgot_pass_button.setEnabled(s.toString().trim().contains("@"));
                } else {
                    forgot_pass_button.setEnabled(false);
                }
            }

            @Override
            public void afterTextChanged(Editable s) {}
        });

        forgot_pass_button.setOnClickListener(v -> {
            if (!con.connectedToInternet()) {
                new MaterialAlertDialogBuilder(this)
                        .setTitle(Constants.noInternetTitle)
                        .setMessage(Constants.noInternetMessage)
                        .setPositiveButton("OK", (dialog, which) -> dialog.dismiss())
                        .create()
                        .show();
            } else {
                FirebaseAuth.getInstance().sendPasswordResetEmail(forgot_editText.getText().toString().trim())
                        .addOnFailureListener(e -> {
                            //Log.d(TAG, "Email send failure");
                            new MaterialAlertDialogBuilder(ForgotPassword.this, R.style.ReportAlertTheme)
                                    .setTitle(Constants.accountNotFound)
                                    .setMessage(Constants.accountNotFoundMessage)
                                    .setCancelable(false)
                                    .setPositiveButton("Ok", (dialog, which) -> dialog.dismiss())
                                    .create()
                                    .show();
                        })
                        .addOnCompleteListener(task -> {
                            if (task.isSuccessful()) {
                                //Log.d(TAG, "Reset password email sent");
                                new MaterialAlertDialogBuilder(ForgotPassword.this, R.style.ReportAlertTheme)
                                        .setTitle(Constants.resetPassTitle)
                                        .setMessage(Constants.resetPassMessage)
                                        .setCancelable(false)
                                        .setPositiveButton("OK", (dialog, which) -> {
                                            dialog.dismiss();
                                            finish();
                                        })
                                        .create()
                                        .show();
                            }
                        });
            }
        });
    }
}

File: app/src/main/java/com/app/scavenger/v2/activities/ForgotPassword.kt
package com.app.scavenger.v2.activities

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import androidx.appcompat.app.AppCompatActivity
import androidx.core.widget.addTextChangedListener
import com.app.scavenger.BaseApplication
import com.app.scavenger.R
import com.app.scavenger.databinding.ActivityForgotPasswordBinding
import com.app.scavenger.dialog.DialogBuilder
import com.app.scavenger.utils.ConnectionDetector
import com.app.scavenger.utils.Constants

class ForgotPassword: AppCompatActivity() {

    private lateinit var binding: ActivityForgotPasswordBinding
    private lateinit var connection: ConnectionDetector

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityForgotPasswordBinding.inflate(layoutInflater)
        setContentView(binding.root)

        connection = ConnectionDetector(this)
        binding.forgotToolbar.setTitle("Forgot Password")
        binding.forgotPassButton.isEnabled = false

        binding.forgotEditText.addTextChangedListener {
            object : TextWatcher {

                override fun beforeTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {}

                override fun onTextChanged(p0: CharSequence?, p1: Int, p2: Int, p3: Int) {
                    if (p0?.toString()?.trim()?.length != 0) {
                        binding.forgotPassButton.isEnabled = p0?.toString()?.trim()?.contains("@") == true
                    } else {
                        binding.forgotPassButton.isEnabled = false
                    }
                }

                override fun afterTextChanged(p0: Editable?) {}
            }
        }

        binding.forgotPassButton.setOnClickListener {
            if (!connection.connectedToInternet()) {
                DialogBuilder.showInformationDialog(
                    this,
                    null,
                    Constants.noInternetTitle,
                    Constants.noInternetMessage
                )
            } else {
                BaseApplication.instance.getFirebaseAuth()
                    .sendPasswordResetEmail(binding.forgotEditText.text.toString().trim())
                    .addOnFailureListener {
                        DialogBuilder.showInformationDialog(
                            this,
                            R.style.ReportAlertTheme,
                            Constants.accountNotFound,
                            Constants.accountNotFoundMessage
                        )
                    }
                    .addOnCompleteListener {
                        if (it.isSuccessful) {
                            DialogBuilder.showErrorDialog(
                                this,
                                R.style.ReportAlertTheme,
                                Constants.resetPassTitle,
                                Constants.resetPassMessage
                            ) { dialog, _ ->
                                dialog.dismiss()
                                finish()
                            }
                        }
                    }
            }
        }
    }
}